common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -32,6 +33,16 @@ def setUp(self):
             self.browser, username=self.USERNAME, password=self.PASSWORD, email=self.EMAIL
         ).visit().get_user_id()
 
+    def assert_event_emitted_num_times(self, setting, num_times):
+        """
+        Verify a particular user settings change event was emitted a certain
+        number of times.
+        """
+        # pylint disable=no-member
+        super(AccountSettingsTestMixin, self).assert_event_emitted_num_times(
```

Comment:
Note, I added support for testing referer in #7694. Perhaps you can just use that, as long as you are willing to change from merely testing the number of events to testing the expected event data.